backend/asmlink.ml

```diff
@@ -230,20 +229,26 @@ let force_linking_of_startup ~ppf_dump =
   Asmgen.compile_phrase ~ppf_dump
     (Cmm.Cdata ([Cmm.Csymbol_address "caml_startup"]))
 
-let make_globals_map units_list ~crc_interfaces =
-  let crc_interfaces = String.Tbl.of_seq (List.to_seq crc_interfaces) in
+let make_globals_map units_list =
+  (* The order in which entries appear in the globals map does not matter
+     (see the natdynlink code).
+     Note that [!interfaces] can contain duplicates.
+     We can corrupt [!interfaces] since it won't be used again until the next
+     compilation. *)
```

Comment:
You can corrupt it, but why bother? It's just as easy to make a new ref initiated to its value and mutate that instead.
Of course, if we switch `interfaces` to a hash table then you probably will need to corrupt it.